include/seastar/core/internal/io_request.hh

```diff
@@ -38,112 +38,244 @@ public:
     enum class operation { read, readv, write, writev, fdatasync, recv, recvmsg, send, sendmsg, accept, connect, poll_add, poll_remove, cancel };
 private:
     operation _op;
-    int _fd;
```

Comment:
Why moving `_fd` out of "global" location? All operations use it

Reply:
yes, all operations use `_fd`. but for the sake of consistency, i am inclined to put all operands of an operation into its own structure. let's think this way, before this change, we try to extract the common parts of operands into a shared union. after this change, all the shared bits are restructured into operation structure itself. `fd` is not an exception in this regard even it is shared by *all* operations.